components/scripts/lib/build-scan-online.sh

```diff
@@ -44,17 +44,53 @@ is_build_scan_metadata_missing() {
   return 0
 }
 
-fetch_and_read_build_scan_data() {
+fetch_and_read_single_build_scan() {
+  local build_scan_url="$1"
+
+  local build_scan_data
+  build_scan_data="$(fetch_build_scan_data brief_logging "${build_scan_url}")"
+
+  debug "Raw build scan data"
+  debug "---------------------------"
+  debug "${build_scan_data}"
```

Comment:
You will see this `build_scan_data` nomenclature in my changes. I don't know if it's quite right because really it might contain more than Build Scan data. It can also contain the build time metrics. It may also possibly contain other data in the future.

Reply:
One way out that I think would be easy to understand and distinguish is:
- _build scan data_ for everything that refers to data derived from a single build scan
- _build scan insights_ for everything that refers to an umbrella of build scan data and and additional metrics

Reply:
I really like this! It is exactly what I was looking for.

Reply:
I think I will hold off on doing this renaming for now. There are other occurrences of this `build_scan_data` nomenclature throughout the codebase, including in places not touched by this PR. There may even be some places we can apply this terminology to in the Java code too. 

I even think [these functions](https://github.com/gradle/gradle-enterprise-build-validation-scripts/pull/383#discussion_r1152658823) would be better renamed as: 

- `fetch_single_build_scan` -> `fetch_build_scan_data`
- `parse_single_build_scan` -> `parse_build_scan_data`
- `fetch_build_scans_and_build_time_metrics` -> `fetch_build_scan_insights`
- `parse_build_scans_and_build_time_metrics` -> `parse_build_scan_insights`

I will add this as a next step to discuss.

Reply:
SGTM